HW0/P4/P4.py

```diff
@@ -40,37 +49,89 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    x_coords, y_coords, z_coords = np.loadtxt('P4_measurements.txt', delimiter=',', unpack=True)
+
+    #create a matrix of the measurements for part 5
+    measurements = np.matrix([x_coords, y_coords, z_coords])
+    
+    x_coords = x_coords/rx
+    y_coords = y_coords/ry
+    z_coords = z_coords/rz    
+    
+
+    ax.plot(x_coords, y_coords, z_coords,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1., 0., 0., dt, 0., 0.], 
+                   [0., 1., 0., 0., dt, 0.],
+                   [0., 0., 1., 0., 0., dt],
+                   [0., 0., 0., 1 - c*dt, 0., 0.],
+                   [0., 0., 0., 0., 1 - c*dt, 0.],
+                   [0., 0., 0., 0., 0., 1 - c*dt]]) 
+    a = np.transpose(np.matrix([0.,0.,0.,0.,0.,g*dt]))
+    s = matlib.zeros((6,K))
 
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    s[:, 0] = np.transpose(np.matrix([0.,0.,2.,15.,3.5,4.]))
+    for x in range(1, K):
+        s[:,x] = A*s[:,x-1] + a
+
+    x_coords = np.asarray(s[0])[0]
+    y_coords = np.asarray(s[1])[0]
+    z_coords = np.asarray(s[2])[0]
+    ax.plot(x_coords, y_coords, z_coords,
+             '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.matrix([[bx, 0., 0., 0., 0., 0.], 
+                   [0., by, 0., 0., 0., 0.],
+                   [0., 0., bz, 0., 0., 0.],
+                   [0., 0., 0., bvx, 0., 0.],
+                   [0., 0., 0., 0., bvy, 0.],
+                   [0., 0., 0., 0., 0., bvz]]) 
+    C = np.matrix([[rx, 0., 0., 0., 0., 0.], 
+                   [0., ry, 0., 0., 0., 0.],
+                   [0., 0., rz, 0., 0., 0.]]) 
+                   
```

Comment:
Completely missed this note when I wrote the above comment, but like I said, I think hardcoding in this case is fine because it shows you at a glance what the matrix actually looks like.